addon/components/universal-selection/component.js

```diff
@@ -0,0 +1,42 @@
+import Component from '@ember/component';
+import EmberObject from '@ember/object';
+import { inject as service } from '@ember/service';
+import RSVP from 'rsvp';
+
+import layout from './template';
+
+export default Component.extend({
+  layout,
+
+  exports: service(),
+  toast: service(),
+
+  _model: '',
+  items: null,
+  placeholder: 'Select a list to import influencers',
+  current: null,
+
+  search(keyword) {
+    return new RSVP.Promise((resolve) => {
```

Comment:
`this.get('exports').searchEntities` itself returns a Promise so why wrap it in another one ?

Reply:
if i just do something like
```
    return this.get('exports').searchEntities(keyword).then((response) => {
        let allItems = [];
        Object.keys(response).forEach(key => {
          response[key].map((item) => {
            allItems.push(
              EmberObject.create({entityType: key, item})
            );
          });
        });
        // resolve(allItems);
        return allItems;
      });
```

then when i search, i get back 'No results' instead of 'Loading', but if i wrap with another promise then i get back 'loading' when it's still fetching the results. maybe im doing something wrong.